runtime/compiler/optimizer/IdiomTransformations.cpp

```diff
@@ -9351,6 +9351,14 @@ CISCTransform2ArrayCmp(TR_CISCTransformer *trans)
          generateArraycmplen = true;
          }
       }
+   if (generateArraycmplen)
+      {
+      if (!comp->cg()->getSupportsArrayCmpLen())
+         {
+         // arrayCmpLen is not supported by codegen
+         generateArraycmplen = false;
```

Comment:
It's not clear to me that we can simply elect not to use `arraycmplen` despite the determination of the above logic. Instead of setting `generateArraycmplen` to false, I think we should consider the length result to be potentially required, and so refuse to transform in this situation

Reply:
Thanks for updating this! It should work, though I did like the previous code structure that tested for support in only one place. Now the `getSupportsArrayCmpLen()` condition is repeated in 4 different places, which to me doesn't read as clearly. I also think it would be good to trace when giving up regardless of why `generateArraycmplen` was set, and the single test you had before would have been a good place to do that. What do you think?